### DISCUSSION TITLE
[Feature] Self learning AI

### DISCUSSION THREAD

**Discussion by Amidweiz** [ikemen-engine/Ikemen-GO#829] [Feature Request]
This is technically already an undocumented feature present within mugen. If you put the 
`AI = whatever.ai` line within a def file in which mugen will write some encrypted form of code to it. Normally there is only a minor benefit from it in mugen in AI matches with it's wins seemingly only increasing by about %50 up against mugen AI without it. There was also meant to be a way to train it via training mode and a setting within the pause menu (which currently isn't implemented within ikemen go plus I think) http://mugenguild.com/forum/topics/yall-know-that-ai-file-is-not-only-placeholder-155808.0.html Another problem with this is despite writing it's code to a file, it is one computer specific where if you happen to use the same AI file on another computer it will override the code with new code.

I was thinking that there could be a modern nural network set up using said file where it could learn from previous match ups with a similar implementation to this https://medium.com/@kidtronnix/making-a-neural-network-from-scratch-using-go-2bfd4dafce9c, way to train it via the pause menu in training mode (when said menu gets implemented), and have it be permanent.

**Comment by CableDorado2** [comment]
Related Showcase: https://youtube.com/watch?v=6pYyfUosdGo
Repo: https://github.com/KDing0/Ikemen-GO
